server/src/main/java/com/hexacore/tayo/reservation/ReservationService.java

```diff
@@ -0,0 +1,155 @@
+package com.hexacore.tayo.reservation;
+
+import com.hexacore.tayo.car.CarRepository;
+import com.hexacore.tayo.car.ImageRepository;
+import com.hexacore.tayo.car.model.CarEntity;
+import com.hexacore.tayo.car.model.ImageEntity;
+import com.hexacore.tayo.car.model.ModelEntity;
+import com.hexacore.tayo.common.DataResponseDto;
+import com.hexacore.tayo.common.ResponseDto;
+import com.hexacore.tayo.common.errors.ErrorCode;
+import com.hexacore.tayo.common.errors.GeneralException;
+import com.hexacore.tayo.reservation.dto.CreateReservationDto;
+import com.hexacore.tayo.reservation.dto.GetGuestReservationsResponseDto;
+import com.hexacore.tayo.reservation.dto.GetHostReservationsResponseDto;
+import com.hexacore.tayo.reservation.model.Guest;
+import com.hexacore.tayo.reservation.model.GuestReservation;
+import com.hexacore.tayo.reservation.model.HostCar;
+import com.hexacore.tayo.reservation.model.HostReservation;
+import com.hexacore.tayo.reservation.model.ReservationEntity;
+import com.hexacore.tayo.reservation.model.ReservationStatus;
+import com.hexacore.tayo.user.model.UserEntity;
+import jakarta.transaction.Transactional;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ReservationService {
+
+    private final ReservationRepository reservationRepository;
+    private final CarRepository carRepository;
+    private final ImageRepository imageRepository;
+
+    @Transactional
+    public ResponseDto createReservation(CreateReservationDto createReservationDto) {
+        long guestUserId = 13L; // TODO: JWT 토큰에서 userId를 추출하고 로그인한 상태에서만 실행되도록 추가
+        // 현재 예시는 Guest 유저의 ID:13
+
+        // TODO: userRepository에서 userEntity 제공받기
+        UserEntity guestUser = UserEntity.builder().id(guestUserId).build();
+
+        CarEntity carEntity = carRepository.findById(createReservationDto.getCarId())
+                .orElseThrow(() -> new GeneralException(ErrorCode.CAR_NOT_FOUND));
+
+        UserEntity hostUser = carEntity.getOwner();
+
+        List<List<LocalDateTime>> possibleRentDates = carEntity.getDates();
+
+        LocalDateTime rentDate = createReservationDto.getRentDate();
+        LocalDateTime returnDate = createReservationDto.getReturnDate();
+
+        possibleRentDates.stream()
+                // date.get(0) ~ date.get(1) 사이의 날짜인지 검증
+                .filter(date -> date.get(0).isBefore(rentDate))
+                .filter(date -> date.get(1).isAfter(returnDate))
+                .findFirst()
+                .orElseThrow(() -> new GeneralException(ErrorCode.RESERVATION_DATE_NOT_IN_RANGE));
```

Comment:
```suggestion
        for (List<LocalDateTime> possibleDate : possibleRentDates) {
                if (possibleDate.get(0).isBefore(rentDate)) {
                     if (possibleDate.get(1).isAfter(returnDate) {
                             break; // 예약 가능한 경우
                     }
                } else {
                        throw new GeneralException(ErrorCode.RESERVATION_DATE_NOT_IN_RANGE);
                }
        }

```
지금은 예를 들어 
1~3, 6~8, 10~12, ... 이런 식으로 예약 가능한 날짜가 있을 때,
4~5 에 대한 예약이 들어오면 사실 6~8 부터는 검사하지 않아도 되는데,
모든 예약 가능한 날짜들에 대해 포함되는지 여부를 확인하는 것 같아요.
이런 식으로 수정하면 더 효율적으로 검사할 수 있을 것 같습니다